core/src/main/java/org/frankframework/http/rest/ApiListenerServlet.java

```diff
@@ -579,9 +579,11 @@ else if(segment.startsWith("{") && segment.endsWith("}")) {
 				 */
 				response.addHeader("Allow", (String) messageContext.get("allowedMethods"));
 
-				if (!Message.isEmpty(result)) {
+				if (!Message.isEmpty(result) && result.size() != -1) {
```

Comment:
This is not the correct change because you now make setting content-type dependent on result-size being known.
What I had suggested earlier was to combine `(!Message.isEmpty(result) || method == ApiListener.HttpMethod.HEAD)`

Checking for size can be done in a nested `if`. And when setting content-length on the response, you should not need to also add a header `content-length` because the HttpResponse implementation should do this for you.